runtime/stdlib/random.go

```diff
@@ -195,3 +206,207 @@ func NewRevertibleRandomFunction(generator RandomGenerator) StandardLibraryValue
 		},
 	)
 }
+
+// cases of a random number of size 8 bytes or less can be all treated
+// by the same function, based on the uint64 type.
+// Although the final output is a `uint64`, it can be safely
+// casted into the desired output type because the extra bytes are guaranteed
+// to be zeros.
+func getUint64RandomNumber(
+	generator RandomGenerator,
+	ty sema.Type,
+	moduloArg interpreter.Value,
+) uint64 {
+	// map for a quick type to byte-size lookup
+	typeToBytes := map[sema.Type]int{
+		sema.UInt8Type: 1, sema.UInt16Type: 2, sema.UInt32Type: 4, sema.UInt64Type: 8,
+		sema.Word8Type: 1, sema.Word16Type: 2, sema.Word32Type: 4, sema.Word64Type: 8,
+	}
```

Comment:
refactor this out of the function, so the map is not re-allocated each time the function is called.
Same below in `getBigRandomNumber`